axelrod/strategies/titfortat.py

```diff
@@ -92,6 +94,39 @@ class TwoTitsForTat(Player):
     def strategy(opponent: Player) -> Action:
         return D if D in opponent.history[-2:] else C
 
+class DynamicTwoTitsForTat(Player):
+    """A player starts by cooperating and then mimics previous move by 
+    opponent with a dynamic bias based off of the opponents ratio of 
+    cooperations to total moves (so their current probability of 
+    cooperating towards cooporating regardless of the move 
+    (aka: forgiveness)."""
+
+    name = 'Dynamic Two Tits For Tat'
+    classifier = {
+        'memory_depth': 2,  # Long memory, memory-2
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    
+    @staticmethod
+    def strategy(opponent):
+		# First move
+        if len(opponent.history) == 0:
+            # Make sure we cooporate first turn
+            return C
+        if D in opponent.history[-2:]:
+            # Probability of turning the other cheek based off of 
+            # opponent's probability of defection
+            if random.random() < (opponent.cooperations / len(opponent.history)):
```

Comment:
Your mock player is in effect a defector there (because `actions=[D, D, D]`. So the player will never cooperate (the probability of it cooperating is 0). Use the mock player you currently have with `actions=[D, C, C, D, C]` (or similar).